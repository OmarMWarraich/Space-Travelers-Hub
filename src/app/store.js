import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
/* eslint-disable-next-line import/prefer-default-export */
export const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
});
