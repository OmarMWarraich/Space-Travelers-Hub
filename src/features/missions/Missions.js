import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Table from 'react-bootstrap/Table';
import { fetchMissionsAsync } from './missionsSlice';
import Mission from './Mission';
import styles from './Mission.module.css';

const Missions = () => {
  const dispatch = useDispatch();

  const missionsList = useSelector((state) => state.missions);

  useEffect(() => {
    if (missionsList.length === 0) {
      dispatch(fetchMissionsAsync());
    }
    /* eslint-disable-next-line react-hooks/exhaustive-deps */
  }, []);

  return (
    <div className={styles.container}>
      <Table bordered hover>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>&nbsp;</th>
          </tr>
        </thead>
        <tbody>
          {missionsList.map((mission) => (
            <Mission key={mission.id} mission={mission} />
          ))}
        </tbody>
      </Table>
    </div>
  );
};

export default Missions;
