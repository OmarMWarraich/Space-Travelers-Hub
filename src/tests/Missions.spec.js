import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import Missions from '../features/missions/Missions';

it('Mission page rendering', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Missions />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
