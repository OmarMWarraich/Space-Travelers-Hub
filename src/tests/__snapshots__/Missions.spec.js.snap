// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission page rendering 1`] = `
<div
  className="container"
>
  <table
    className="table table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Â 
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
